runtime/src/registry.rs

```diff
@@ -345,3 +287,82 @@ decl_event!(
         CheckpointSet(ProjectId, CheckpointId),
     }
 );
+
+/// DecodeKey trait
+///
+/// A DecodeKey type must implement the decode_key function and
+/// it is free to choose what decoding algorithm it prefers.
+///
+/// DecodedKey::Key is the expected decoded key type
+///
+pub trait DecodeKey {
+    type Key: parity_scale_codec::Decode;
+
+    /// Decode the given raw storage map `key`. It's the inverse
+    /// of [frame_support::generator::StorageMap::storage_map_final_key],
+    /// so applying `decode_key` after `storage_map_final_key` must yield
+    /// identity as to the original input key.
+    fn decode_key(key: &[u8]) -> Result<Self::Key, parity_scale_codec::Error>;
+}
+
+impl DecodeKey for store::Orgs {
+    type Key = OrgId;
+
+    fn decode_key(key: &[u8]) -> Result<OrgId, parity_scale_codec::Error> {
+        decode_blake_two128_concat_key(key)
+    }
+}
+
+impl DecodeKey for store::Projects {
+    type Key = ProjectId;
+
+    fn decode_key(key: &[u8]) -> Result<ProjectId, parity_scale_codec::Error> {
+        decode_blake_two128_concat_key(key)
+    }
+}
+
+/// Decode a blake_two128_concat hashed key to the inferred type K.
+///
+/// See [frame_support::storage::StorageMap::storage_value_final_key]
+/// for background on `storage_key_prefix` being 32;
+///
+/// See [substrate_primitives::hashing::blake2_128] for background
+/// on `hashed_key_length` being 16;
+pub fn decode_blake_two128_concat_key<K: parity_scale_codec::Decode>(
+    key: &[u8],
+) -> Result<K, parity_scale_codec::Error> {
+    let storage_key_prefix = 32;
+    let hashed_key_length = 16;
+    let final_storage_key_length = storage_key_prefix + hashed_key_length;
+    let mut id_bytes = &key[final_storage_key_length..];
+    K::decode(&mut id_bytes)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    /// This specific import is required over [frame_support::storage::StorageMap] to build.
+    use crate::sp_api_hidden_includes_construct_runtime::hidden_include::storage::generator::StorageMap;
```

Comment:
This is what the compiler suggests. But it is only an alias for the following
```suggestion
    use frame_support::storage::generator::StorageMap;
```

Reply:
It didn't _build_ with the `frame_support` but I will give it another spin!

Reply:
For the record, it needs the "generator" part (as suggested). I was using `frame_support::storage::StorageMap`. Solved.